spec/autoflow-spec.coffee

```diff
@@ -111,6 +111,16 @@ describe "Autoflow package", ->
         words
       """
 
+    it "does not change the cursor screen position after reflowing paragraphs", ->
```

Comment:
I think another important test case is if the line is `10 * PreferredLineLength` characters long but the cursor is at `[0, 5 * PreferredLineLength + 10]`. Given the problem statement, the cursor should be at the same position within the text, even though the text is now on the sixth buffer line.